### ISSUE TITLE
Creating PrimaryKeys constraint get different constraint names

### ISSUE THREAD

**Issue by Sherif-Ahmed** [fluentmigrator/fluentmigrator#1800]
**Describe the bug**
Creating PrimaryKeys constraint get different constraint names

**To Reproduce**
I have an initial migration for table creation
```
public class InitialMigration : ForwardOnlyMigration
{
    public override void Up()
    {
        Create.Table("MyTable")
            .WithColumn("Id").AsInt32().NotNullable().PrimaryKey()
            .WithColumn("Name").AsString(400).NotNullable();
    }
}
```

**This will generate a constraint name "PK_MyTable"**, and in a later migration I have to re-create the primary key because of a table rename
```
public class InitialMigration : ForwardOnlyMigration
{
    public override void Up()
    {
        Delete.PrimaryKey("PK_MyTable").FromTable("MyTable");
        Rename.Table("MyTable").To("NewTableName");
        Create.PrimaryKey().OnTable("NewTableName").Column("Id");
    }
}
```

**This will generate a constraint name "PK_NewTableName_Id"**
Is this intended or I had an issue.

**Expected behavior**
Same constraint name convention for example "PK_NewTableName" based on  `DefaultPrimaryKeyNameConvention`

**Information (please complete the following information):**
 - OS: Windows 11
 - Platform: .NET 8.0.4
 - FluentMigrator: 5.2.0 "in-process runner"
 - Database Management System: SQLServer

**Comment by TheSquidCombatant** [comment]
Obviously, the observed difference in the name of the primary key occurs due to the fact that, when creating a table, the `DefaultConstraintNameConvention` class is used to generate the name, and when creating a constraint, the `DefaultPrimaryKeyNameConvention` class is used to generate the name.

I agree that it would be correct for the name for the primary key to be generated the same in all cases. I have created a pull request with the appropriate fix, please check https://github.com/fluentmigrator/fluentmigrator/pull/1803.

I won’t reassure you and I’ll say right away that I’m not a maintainer. For now, I suggest keeping in mind that the default primary key name is not guaranteed. And each time explicitly set the name of the primary key.

**Comment by jzabroski** [comment]
This would be a breaking change, but seems reasonable. For now you can implement a different naming convention as a workaround.

**Comment by jzabroski** [comment]
The root cause here is that we're using CreateConstraintExpression.

https://github.com/fluentmigrator/fluentmigrator/blob/7efff096dca439a5cf87adcb58da85c9dfc06f6e/src/FluentMigrator/Builders/Create/CreateExpressionRoot.cs#L129-L144

It gets generated here:

https://github.com/fluentmigrator/fluentmigrator/blob/7efff096dca439a5cf87adcb58da85c9dfc06f6e/src/FluentMigrator.Runner.Core/Generators/Generic/GenericGenerator.cs#L193-L208

Which in turn delegates ultimately to `ColumnBase.GetPrimaryKeyConstraintName`:

https://github.com/fluentmigrator/fluentmigrator/blob/7efff096dca439a5cf87adcb58da85c9dfc06f6e/src/FluentMigrator.Runner.Core/Generators/Base/ColumnBase.cs#L318-L328

1. ColumnBase would need a dependency on IConventionSet to get the primary key name convention 
2. https://github.com/fluentmigrator/fluentmigrator/blob/7efff096dca439a5cf87adcb58da85c9dfc06f6e/src/FluentMigrator.Runner.Core/Conventions/DefaultPrimaryKeyNameConvention.cs#L27 should have an explicit IPrimaryKeyNameConvention interface, although probably not strictly required, it does help with discoverability.
